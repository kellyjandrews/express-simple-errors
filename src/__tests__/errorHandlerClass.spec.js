import ErrorHandler from '../index';

describe('ErrorHandler Class', () => {
  const errors = new ErrorHandler();
  const defaultHandler = errors.handlers.get('Default').toString();
  it('returns default class', () => {
    expect(typeof errors.setHandler).toBe('function');
    expect(typeof errors.unsetHandler).toBe('function');
    expect(typeof errors.handleNotFound).toBe('function');
    expect(typeof errors.handleError).toBe('function');
    expect(typeof errors.sendResponse).toBe('function');
    expect(typeof errors.middleware).toBe('function');
    expect(defaultHandler).toBeDefined();
    expect(errors).toMatchSnapshot();
  });

  it('allows turning off stack trace from response.', () => {
    const errorsStack = new ErrorHandler({stackTrace: false});
    expect(() => (new ErrorHandler({stackTrace: 123}))).toThrow();
    expect(() => (new ErrorHandler({stackTrace: '123'}))).toThrow();
    expect(() => (new ErrorHandler({stackTrace: () => {}}))).toThrow();
    expect(() => (new ErrorHandler({stackTrace: []}))).toThrow();
    expect(() => (new ErrorHandler({stackTrace: {}}))).toThrow();
    expect(errorsStack).toMatchSnapshot();
  });

  it('allows Default handler to be changed', () => {
    errors.setHandler('Default', () => {});
    expect(errors.handlers.get('Default').toString()).toEqual('function () {}');
  });

  it('restores Default handler when unset', () => {
    errors.unsetHandler('Default');
    expect(errors.handlers.get('Default').toString()).toEqual(defaultHandler);
  });

  it('allows custom handler creation', () => {
    expect(() => (errors.setHandler())).toThrow();
    expect(() => (errors.setHandler('Test'))).toThrow();
    expect(() => (errors.setHandler('Test', true))).toThrow();
    expect(() => (errors.setHandler('Test', {}))).toThrow();
    expect(() => (errors.setHandler('Test', []))).toThrow();
    expect(() => (errors.setHandler('Test', 123))).toThrow();
    expect(() => (errors.setHandler('Test', '123'))).toThrow();
    errors.setHandler('Test', () => {});
    expect(errors.handlers).toMatchSnapshot();
  });

  it('allows custom handler deletion', () => {
    expect(() => (errors.unsetHandler())).toThrow();
    errors.unsetHandler('Test');
    expect(errors.handlers).toMatchSnapshot();
  });
});
