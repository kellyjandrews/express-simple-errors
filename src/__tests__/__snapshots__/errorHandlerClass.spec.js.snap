exports[`ErrorHandler Class allows custom handler creation 1`] = `
Array [
  Array [
    "Default",
    [Function],
  ],
  Array [
    "Test",
    [Function],
  ],
]
`;

exports[`ErrorHandler Class allows custom handler deletion 1`] = `
Array [
  Array [
    "Default",
    [Function],
  ],
]
`;

exports[`ErrorHandler Class allows turning off stack trace from response. 1`] = `
ErrorHandler {
  "handlers": Array [
    Array [
      "Default",
      [Function],
    ],
  ],
  "stackTrace": false,
}
`;

exports[`ErrorHandler Class returns default class 1`] = `
ErrorHandler {
  "handlers": Array [
    Array [
      "Default",
      [Function],
    ],
  ],
  "stackTrace": true,
}
`;
